### PR TITLE
Add additional details for diagnosing OMI provider failures.

### PR DESCRIPTION
**Pull request by jeffaco**
@Microsoft/omi-devs 

Please review. This document is a good starting point for diagnosing OMI problems in general, and provider bugs specifically. It also details the information that Microsoft needs in order to diagnose OMI problems (either in OMI itself or in OMI providers).

### CODE REVIEW

File: Unix/doc/diagnose-omi-problems.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+- [Log file entries of interest](#log-file-entries-of-interest)
+- [Required information when reporting problems](#required-information-when-reporting-problems)
+- [Diagnosing memory problems in OMI itself](#diagnosing-memory-problems-in-omi-itself)
+- [Diagnosing OMI Provider Failures](#diagnosing-omi-provider-failures)
+  - [Diagnosing OMI Provider Memory Leaks](#diagnosing-omi-provider-memory-leaks)
+  - [Diagnosing Unexpected OMI Provider aborts](#diagnosing-unexpected-omi-provider-aborts)
+
+### Minimum OMI Version
+
+**NOTE: This document assumes OMI v1.4.1-0 or later.** If you have a
```
Comment (palladia) [inline Unix/doc/diagnose-omi-problems.md]: I know you discuss how to get version number later.  Either move that part up here, or note here that you'll describe how to get version number later.

File: Unix/doc/diagnose-omi-problems.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+easy to retrieve mechanism. This will be resolved in a future OMI
+release. However, we do log the following message when a request is
+completely processed:
+
+```
+2017/11/21 15:44:02 [79169,79169] INFO: null(0): EventId=40011 Priority=INFO done with receiving msg(0x151d218:4099:EnumerateInstancesReq:3)
+```
+
+If you don't see an entry like the above (note that PID is shown in
+angle brackets), that likely indicates a crash or abnormal abort of
```
Comment (palladia) [inline Unix/doc/diagnose-omi-problems.md]: Do't see angle brackets (<>).  Did you mean square brackets([])?

File: Unix/doc/diagnose-omi-problems.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+### Required information when reporting problems
+
+In general, for Microsoft to diagnose OMI or OMI provider problems, we need the
+following pieces of information:
+
+1. The OMI version. Show us output from the command `/opt/omibin/omiserver -v`.
+2. An exact description of the problem you are experiencing,
+3. All log files from `/var/opt/omi/log`, with logging set to INFO mode
+   (or above), when the problem was replicated.
+
```
Comment (palladia) [inline Unix/doc/diagnose-omi-problems.md]: 4.  If possible, steps to reproduce the problem.

File: Unix/doc/diagnose-omi-problems.md (hunk 1 above)
Comment (jeffaco) [reply Unix/doc/diagnose-omi-problems.md]: I copied it so it's in both places.

File: Unix/doc/diagnose-omi-problems.md (hunk 2 above)
Comment (jeffaco) [reply Unix/doc/diagnose-omi-problems.md]: Yup, fixed, thanks for catching this.

File: Unix/doc/diagnose-omi-problems.md (hunk 3 above)
Comment (jeffaco) [reply Unix/doc/diagnose-omi-problems.md]: Added.

File: Unix/doc/diagnose-omi-problems.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+The following sections provide far more detail on exactly what Microsoft
+will need for problem analysis, and how to provide us with that information.
+
+## Enabling logging in OMI
+
+Currently, OMI requires logging set to the `INFO` level or higher in
+order to diagnose problems. Since OMI's
+`/etc/opt/omi/conf/omiserver.conf` file doesn't currently have the
+logging entry in the configuration file, add the following lines to
+`/etc/opt/conf/omiserver.conf`:
```
Comment (JumpingYang001) [inline Unix/doc/diagnose-omi-problems.md]: a minor issue: the path should be /etc/opt/omi/conf/omiserver.conf , thanks.

**Pull request merged**